test/cluster/functionality.sh

```diff
@@ -43,41 +56,45 @@ test_token="$( oc whoami -t )"
+# We can reach the Elasticsearch service at serviceName:apiPort
+elasticsearch_api="$( oc get svc "${OAL_ELASTICSEARCH_SERVICE}" -o jsonpath='{ .metadata.name }:{ .spec.ports[?(@.targetPort=="restapi")].port }' )"
 
 for kibana_pod in $( oc get pods --selector component="${OAL_KIBANA_COMPONENT}"  -o jsonpath='{ .items[*].metadata.name }' ); do
 	os::log::info "Testing Kibana pod ${kibana_pod} for a successful start..."
-	os::cmd::try_until_text "oc logs ${kibana_pod} -c kibana" "Server running at http://0\.0\.0\.0:5601" "$(( 10*TIME_MIN ))"
-	os::cmd::try_until_text "oc logs ${kibana_pod} -c kibana" "Kibana index ready" "$(( 10*TIME_MIN ))"
+	os::cmd::try_until_text "oc exec ${kibana_pod} -c kibana -- curl -s --request HEAD --write-out '%{response_code}' http://localhost:5601/" "200" "$(( 10*TIME_MIN ))"
+	os::cmd::expect_success_and_text "oc get pod ${kibana_pod} -o jsonpath='{ .status.containerStatuses[?(@.name==\"kibana\")].ready }'" "true"
+	os::cmd::expect_success_and_text "oc get pod ${kibana_pod} -o jsonpath='{ .status.containerStatuses[?(@.name==\"kibana-proxy\")].ready }'" "true"
 done
 
-for elasticsearch_pod in $( oc get pods --selector component="${OAL_ELASTICSEACH_COMPONENT}" -o jsonpath='{ .items[*].metadata.name }' ); do
-	os::log::info "Testing ElasticSearch pod ${elasticsearch_pod} for a successful start..."
-	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[cluster\.service\s*\]" "$(( 10*TIME_MIN ))"
-	os::cmd::try_until_text "oc logs ${elasticsearch_pod}" "\[node\s*\]\s*\[.*\]\s*started" "$(( 10*TIME_MIN ))"
+for elasticsearch_pod in $( oc get pods --selector component="${OAL_ELASTICSEARCH_COMPONENT}" -o jsonpath='{ .items[*].metadata.name }' ); do
+	os::log::info "Testing Elasticsearch pod ${elasticsearch_pod} for a successful start..."
+	os::cmd::try_until_text "curl_es '${elasticsearch_pod}' '/' -X HEAD -w '%{response_code}'" '200' "$(( 10*TIME_MIN ))"
+	os::cmd::expect_success_and_text "oc get pod ${elasticsearch_pod} -o jsonpath='{ .status.containerStatuses[?(@.name==\"elasticsearch\")].ready }'" "true"
 
-	os::log::info "Checking that ElasticSearch pod ${elasticsearch_pod} recovered its indices after starting..."
-	if oc logs "${elasticsearch_pod}" | grep -E "\[cluster\.service\s*\]" | grep -q "new_master"; then
+	os::log::info "Checking that Elasticsearch pod ${elasticsearch_pod} recovered its indices after starting..."
+	os::cmd::try_until_text "curl_es '${elasticsearch_pod}' '/_cluster/state/master_node' -X HEAD -w '%{response_code}'" '200' "$(( 10*TIME_MIN ))"
+	es_master_id="$( curl_es "${elasticsearch_pod}" "/_cluster/state/master_node" | jq --raw-output '.master_node' )"
+	es_pod_node_id="$( curl_es "${elasticsearch_pod}" "/_nodes/_local" | jq '.nodes' | jq --raw-output 'keys[0]' )"
+	if [[ "${es_master_id}" == "${es_pod_node_id}" ]]; then
 		os::cmd::expect_success_and_text "oc logs ${elasticsearch_pod}" "\[gateway\s*\]\s*\[.*\]\s*recovered\s*\[[0-9]*\]\s*indices into cluster_state"
```

Comment:
Not certain but I think that's the message you see when the node has finished startup (which includes recovery after an outage) and is ready for use. If it hasn't logged this then it's not really ready. Not sure if there's a health check for readiness on the ES API?